src/Websites/Websites/Utilities/WebsitesClient.cs

```diff
@@ -112,6 +112,9 @@ public void UpdateWebApp(string resourceGroupName, string location, string webAp
 
             if (siteEnvelope != null)
             {
+                System.Reflection.PropertyInfo property = siteEnvelope.GetType().GetProperty("VnetInfo");
```

Comment:
Is your purpose to find a specific sub-object of class Site which has VnetInfo property or all sub-objects which have this property?

Reply:
Actually, the VnetInfo property is not part of the Site class. It has been added to wrapper class [PSSite ](https://github.com/Azure/azure-powershell/blob/main/src/Websites/Websites/Models.WebApp/PSSite.cs#L93)to fetch VnetInfo for the webapp. When the PSSite object is used to update the webapp, it throws the deserialization exception as it expects an object of type Site and it does not have a VnetInfo property. So I am checking the existence of VnetInfo and making it null before deserializing.

Reply:
PSSite is a child class of Site. If in the future, another child class which also has the VnetInfo property, would this piece of code have problem? So my point is would you check if siteEnvelop is an instance of PSSite, and then you can access the VnetInfo directly after converting the siteEnvelop object to an instance of PSSite? Or if any siteEnvelop instance which has that property, you would set its value as null.

Reply:
updated the code as per the review comments.